android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java

```diff
@@ -181,13 +299,23 @@ public BraveNewTabPageLayout(Context context, AttributeSet attrs) {
     @Override
     protected void onFinishInflate() {
         super.onFinishInflate();
+
         ntpWidgetLayout = findViewById(R.id.ntp_widget_layout);
         indicatorLayout = findViewById(R.id.indicator_layout);
         ntpWidgetViewPager = findViewById(R.id.ntp_widget_view_pager);
         ntpWidgetAdapter = new NTPWidgetAdapter();
         ntpWidgetAdapter.setNTPWidgetListener(ntpWidgetListener);
         ntpWidgetViewPager.setAdapter(ntpWidgetAdapter);
 
+        SharedPreferences sharedPreferences = ContextUtils.getAppSharedPreferences();
+
+        isNewsOn = BravePrefServiceBridge.getInstance().getNewsOptIn();
+        isShowOptin = sharedPreferences.getBoolean(BraveNewsPreferences.PREF_SHOW_OPTIN, false);
+        isShowNewsOn = BravePrefServiceBridge.getInstance().getShowNews();
```

Comment:
shouldn't that go under `if (ChromeFeatureList.isEnabled(BraveFeatureList.BRAVE_NEWS)){`?

Reply:
done